samples/caf_sensor_manager/configuration/nrf5340dk_nrf5340_cpuapp/event_proxy_init.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#define MODULE event_proxy_init
+
+#include <ipc/ipc_service.h>
+#include <event_manager_proxy.h>
+#include <caf/events/sensor_data_aggregator_event.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(MODULE);
+
+int init_event_proxy(void)
+{
+	const struct device *ipc_instance  = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+	int ret;
+
+	ret = event_manager_proxy_add_remote(ipc_instance);
+	if (ret) {
+		LOG_ERR("Cannot add remote: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return ret;
+	}
+	LOG_INF("Event proxy remote added");
+
+	ret = EVENT_MANAGER_PROXY_SUBSCRIBE(ipc_instance,
+					sensor_data_aggregator_release_buffer_event);
+	if (ret) {
+		LOG_ERR("Cannot register to remote sen_data_agg_release_buffer_event: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return ret;
+	}
+	LOG_INF("Event proxy sensor_data_aggregator_release_buffer_event registered");
+
+	ret = event_manager_proxy_start();
+	if (ret) {
+		LOG_ERR("Cannot start event manager proxy: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return ret;
+	}
+	LOG_INF("Event manager proxy started");
+
+	/* Wait for all the remote cores to finish event proxy initialization. */
+	ret = event_manager_proxy_wait_for_remotes
+		(K_MSEC(CONFIG_SAMPLE_SENSOR_MANAGER_REMOTE_CORE_INITIALIZATION_TIMEOUT));
+	if (ret) {
+		LOG_ERR("Error when waiting for remote: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return ret;
+	}
+	LOG_INF("All remotes ready");
+
+	LOG_INF("Initialization finished successfully");
+	return ret;
+}
+
+
+int init_event_proxy_remote(void)
```

Comment:
I expect these two function should execute on different cores - don't place them in the same file